huaweicloud/services/cbh/data_source_huaweicloud_cbh_flavors.go

```diff
@@ -119,7 +127,18 @@ func datasourceFlavorsRead(_ context.Context, d *schema.ResourceData, meta inter
 	getCBHFlavorsPath := getCBHFlavorsClient.Endpoint + getCBHFlavorsHttpUrl
 	getCBHFlavorsPath = strings.ReplaceAll(getCBHFlavorsPath, "{project_id}",
 		getCBHFlavorsClient.ProjectID)
-	getCBHFlavorsPath += "?action=create"
+
+	// If the `action` is set to **update**, the `spec_code` parameter is required.
+	// If the `action` is not filled in, set it to **create** query.
+	action := d.Get("action")
+	if action != "" {
+		getCBHFlavorsPath += "?action=" + fmt.Sprintf("%v", action)
+		if action == "update" {
+			getCBHFlavorsPath += "&spec_code=" + fmt.Sprintf("%v", d.Get("spec_code"))
+		}
+	} else {
+		getCBHFlavorsPath += "?action=create"
+	}
```

Comment:
Write a separate method to splice query parameters.

Reply:
done.